### DISCUSSION TITLE
Customize responses using databucket

### DISCUSSION THREAD

**Discussion by notfounnd** [mockoon/mockoon#1408] [Ideas] [answered]
I'd like to know if have any way to customize the response using databucket.

E.g.:

Databucket/request model:

```
{
  field_1: data_1
  field_2: data_2
  field_3: data_3
  field_4: data_4
}
```

Response model:

```
{
  field_3: data_3
  hardcode_field: some_message
}
```

**Comment by 255kb** [comment]
Not out of the box. You would have to create a template manually that reuses part of the bucket and mix it with some other properties. It's doable with Handlebars but not very straightforward. 
Let's say you have a bucket named `data1` containing what you showed in your question, you could do something like this: 

```json
{
    "field_3": "{{data 'data1' 'field_3'}}"
    "hardcode_field": "some_message"
}
```

But if you use a CRUD route this won't be possible as the response is automatically generated from the data bucket.

**Comment by notfounnd** [comment]
Thank you @255kb.

If as possible, I would like to suggest an increment to the CRUD feature.

I'm using Mockoon following the same way that you explained, but in this case the dummy data reuse stay limited.

Increment suggest:

- Add a textarea field in response option when databucket is selected. This field would allow the possibility of editing the response struct before Mockoon returns the response to the requester.

I belive this increment will can optimize the CRUD feature use, making its behavior closer to some other real scenarios that exists.

![image](https://github.com/mockoon/mockoon/assets/57534340/e40e0cc0-c7c4-43f0-8b60-c0b61d053b7b)

**Reply by 255kb** [reply, marked as answer]
This could be useful indeed. 
You can open an issue and see if it gather some interest. 

Thanks

**Reply by notfounnd** [reply]
Done in https://github.com/mockoon/mockoon/issues/1419 issue.
